hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java

```diff
@@ -489,6 +532,29 @@ private boolean innerDelete(Path f, boolean recursive) throws IOException {
     }
   }
 
+  /**
+   * Deletes the children of the bucket i.e all files and dirs under it's
+   * path by iterating though th DeleteIteratorWithFSO.
+   *
+   * @param f directory path to be deleted
+   * @return true if successfully deletes all required keys, false otherwise
+   * @throws IOException
+   */
+  private boolean recursiveBucketDelete(Path f, boolean recursive)
+      throws IOException {
+    LOG.trace("delete() path:{}", f);
+    try {
+      DeleteIteratorWithFSO iterator = new DeleteIteratorWithFSO(f, recursive);
```

Comment:
This way we can avoid multiple rpc batch deleteObjects() calls and make the implementation simpler without having `**WithFSO` specific classes, isn't it?

Reply:
yes, this seems like a good idea, we need to handle this in server now. Thanks!

Reply:
Adding hint based on online discussion with @sadanand48. Here the new ofs client can set `recursive=true` and at the server side `OMBucketDeleteRequest` can add logic to handle the `omMetadataManager.isBucketEmpty` check. 
```
message DeleteBucketRequest {
    required string volumeName = 1;
    required string bucketName = 2;
    optional bool recursive = 3 [default = false];
}
```

Reply:
Ozone doesn't currently support recursive bucket deletion by design. Is this needed for hdfs compatibility in FSO buckets? This field is being added to the general bucket delete request so we might want to be careful how we handle this field on existing object store buckets.

Reply:
Thanks @errose28 for the review.
Yes currently there exists no recursive bucketDelete api call to the OM ,however we can set the default value of recursive to false so that no operations on object store buckets are affected. The only use case is this particular case of HDDS-6414 i.e for FSO buckets. 
We are choosing this approach only because the approach in the current PR would create multiple RPC's for the client on a bucket delete thereby potentially blocking it and the new approach would be an optimized one with just one delete call. 
@errose28 , Please let me know if the approach sounds good.
Anyway I'll create a new PR for the second approach in case anybody has objections .

Reply:
Changing this to be handled on the server side in a follow-up PR like you said sounds good.

> This field is being added to the general bucket delete request so we might want to be careful how we handle this field on existing object store buckets

Since bucket delete with recursive = true on a non-FSO bucket is the only case where this flag does not make sense, a not supported operation error from the server seems logical here.

Reply:
Thanks for the discussion around this. On a second thought, I agreed to avoid `recursive` flag attribute in the proto with additional special handling on the server side. Probably, can proceed with the existing deleteObjects() call by sending immediate sub-paths for FSO bucket. We can revisit if there is any performance requirement comes or any other case.